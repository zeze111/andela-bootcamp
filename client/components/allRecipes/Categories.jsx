import React from 'react';
import PropTypes from 'prop-types';
import { Dropdown } from 'react-materialize';

const Categories = (props) => {
  const appetizer = 'Appetizer';
  const main = 'Main';
  const dessert = 'Dessert';
  const drinks = 'Drinks';
  const all = 'All Recipes';
  return (
    <Dropdown
      className="content-recipes"
      trigger={
        <div className="div-pointer">
          <h5 className="light top caps2">
            {props.dropdown || all}
            <i className="material-icons">arrow_drop_down</i>
          </h5>
        </div>
                }
    >
      <ul>
        <li className="grey-text default-droplist">
          <div>Select Category</div>
        </li>
        <li>
          <button
            className="dropdown-btn left-align white waves-effect waves-grey"
            onClick={() => props.onSelectAllRecipes(all)}
          >{all}
          </button>
        </li>
        <li>
          <button
            className="dropdown-btn left-align white waves-effect waves-grey"
            onClick={() => props.onSelectCategory(appetizer)}
          >{appetizer}
          </button>
        </li>
        <li>
          <button
            className="dropdown-btn left-align white waves-effect waves-grey"
            onClick={() => props.onSelectCategory(main)}
          >{main}
          </button>
        </li>
        <li>
          <button
            className="dropdown-btn left-align white waves-effect waves-grey"
            onClick={() => props.onSelectCategory(dessert)}
          >{dessert}
          </button>
        </li>
        <li>
          <button
            className="dropdown-btn left-align white waves-effect waves-grey"
            onClick={() => props.onSelectCategory(drinks)}
          >{drinks}
          </button>
        </li>
      </ul>
    </Dropdown>
  );
};

Categories.propTypes = {
  onSelectAllRecipes: PropTypes.func.isRequired,
  onSelectCategory: PropTypes.func.isRequired,
  dropdown: PropTypes.string.isRequired
};

export default Categories;
