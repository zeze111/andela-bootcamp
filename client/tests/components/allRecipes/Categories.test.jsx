import React from 'react';

import Categories from '../../../components/allRecipes/Categories';
import store from '../../../store';

const props = {
  onSelectAllRecipes: jest.fn(() => Promise.resolve()),
  onSelectCategory: jest.fn(() => Promise.resolve()),
  dropdown: 'Main'
};

const setup = () => {
  const wrapper = shallow(<Categories {...props} store={store} />);
  return { wrapper }
}

describe('onSelectAllRecipes', () => {
  it('select all dropdown category', () => {
    const { wrapper } = setup();
    const event = 'all';
    const button = wrapper.find('#all-button')

    button.simulate('click', event);
  });
});

describe('onSelectCategory', () => {
  it('select main dropdown category', () => {
    const { wrapper } = setup();
    const event = 'Main';
    const button = wrapper.find('#main-button')

    button.simulate('click', event);
  });
});

describe('onSelectCategory', () => {
  it('select appetizer dropdown category', () => {
    const { wrapper } = setup();
    const event = 'Appetizer';
    const button = wrapper.find('#app-button')

    button.simulate('click', event);
  });
});

describe('onSelectCategory', () => {
  it('select drinks dropdown category', () => {
    const { wrapper } = setup();
    const event = 'Drinks';
    const button = wrapper.find('#drink-button')

    button.simulate('click', event);
  });
});

describe('onSelectCategory', () => {
  it('select dessert dropdown category', () => {
    const { wrapper } = setup();
    const event = 'Dessert';
    const button = wrapper.find('#dess-button')

    button.simulate('click', event);
  });
});

describe('Test for Categories', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
