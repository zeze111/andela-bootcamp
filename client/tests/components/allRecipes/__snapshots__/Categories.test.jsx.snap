// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Categories should render component 1`] = `
Object {
  "wrapper": <Dropdown
    className="content-recipes"
    trigger={
      <div
        className="div-pointer"
      >
        <h5
          className="light top text-headers caps2"
        >
          Main
          <i
            className="material-icons"
          >
            arrow_drop_down
          </i>
        </h5>
      </div>
    }
  >
    <ul>
      <li
        className="grey-text default-droplist"
      >
        <div>
          Select Category
        </div>
      </li>
      <li>
        <button
          className="dropdown-btn left-align white waves-effect waves-grey"
          id="all-button"
          onClick={[Function]}
        >
          All Recipes
        </button>
      </li>
      <li>
        <button
          className="dropdown-btn left-align white waves-effect waves-grey"
          id="app-button"
          onClick={[Function]}
        >
          Appetizer
        </button>
      </li>
      <li>
        <button
          className="dropdown-btn left-align white waves-effect waves-grey"
          id="main-button"
          onClick={[Function]}
        >
          Main
        </button>
      </li>
      <li>
        <button
          className="dropdown-btn left-align white waves-effect waves-grey"
          id="dess-button"
          onClick={[Function]}
        >
          Dessert
        </button>
      </li>
      <li>
        <button
          className="dropdown-btn left-align white waves-effect waves-grey"
          id="drink-button"
          onClick={[Function]}
        >
          Drinks
        </button>
      </li>
    </ul>
  </Dropdown>,
}
`;
